minifiers/terser.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const fingerprinter = require('fingerprinting');
+const Config = require('./config.js');
+const Terser = require('terser');
+
+/**
+ * Minify JS using Terser.
+ *
+ * @param {Object} options Content (compiled JS) and filepath.
+ * @param {Function} done Completion callback.
+ * @returns {Undefined} Return early on minification error.
+ * @public
+ */
+module.exports = function terser(options, done) {
+  const filename = options.file.replace('.js', '.min.js');
+  const config = new Config({
+    minify: options.minify,
+    sourceMapContent: options.map,
+    filename
+  });
+
+  const results = Terser.minify(options.content.toString('utf-8'), config.terser);
```

Comment:
Should the minifiers be a little more abstracted, they're basically the same other than this line. Feels like at least the fingerprinting should get pulled out, probabkly also the filename/config stuff.

Reply:
Yeah I can do that, technically it could just be a single module like `minifyJS` which runs either `terser` or `uglifyjs`. Will not go to that length though as Terser being a hard fork might go a different route programmatically.

Reply:
I'm leaving out the fingerprint, it requires the result of the minifier which would just be replacing a line for other line of JS.